### PR TITLE
SasAudio: Always reinitialize the VAG decoder on sceSasSetVoice, even if already playing

### PR DESCRIPTION
**Pull request by hrydgard**
Fixes terrible noises in After Burner.

~~EDIT: Also breaks the engine noise in Ridge Racer, so this needs more work.~~

Not sure why the logic was like it was, blame history wasn't much clue either.

Will of course test a large number of other games before merging.

### CODE REVIEW

**Comment by hrydgard** [comment]
Oops, breaks the engine noise in Ridge Racer. There is something to this check..

**Comment by unknownbrackets** [comment]
Should it still set playing = on if it's not VAG, like noise or etc.?  Maybe the engine noise uses that?

And should it really restart VAG on every pitch change?  That seems wrong.

-[Unknown]

**Comment by hrydgard** [comment]
It is indeed wrong to reset the VAG on just a pitch change, but, the fact that After Burner sounds right with this change kinda confirms that some detail of this is likely the cause of its problem, at least. I think this will just need a small extension of the old check to work correctly.

**Comment by unknownbrackets** [comment]
Is it enough to ignore `v.ChangedParams(vagAddr == prevVagAddr);` this condition?  I wonder what that was added for, but it's been there a long time and I'm not sure if it's verified in tests or not:

https://github.com/hrydgard/ppsspp/commit/4deaec892446a249e08c038a4b4333b9abfb03f8

Maybe it's just that every set voice restarts it?  That would make more sense than resetting on pitch.

And that pitch thing looks pretty suspect too:

https://github.com/hrydgard/ppsspp/commit/7346c4a042eac8d85a03ab00b026eb5b5929024d

I guess this is really asking for more VAG tests...

-[Unknown]

**Comment by hrydgard** [comment]
Oh I didn't even notice that ChangedParams was called from SetPitch as well. That's likely part of why Ridge Racer broke - reinitializing the vag decoder on solely a pitch change is just wrong indeed. It seems that the part of ChangedParams that is useful from there is basically  `if (!playing && on) playing = true`. Which may or may not itself be accurate..

Yeah, tests would indeed be good, heh.

**Comment by hrydgard** [comment]
I simply got rid of the confusing ChangedParams thing. Now both Ridge Racer and After Burner work great. Will test some more games.

File: Core/HLE/sceSas.cpp [hunk 1]
```diff
@@ -363,8 +363,9 @@ static u32 sceSasSetVoice(u32 core, int voiceNum, u32 vagAddr, int size, int loo
 	v.type = VOICETYPE_VAG;
 	v.vagAddr = vagAddr;  // Real VAG header is 0x30 bytes behind the vagAddr
 	v.vagSize = size;
-	v.loop = loop ? true : false;
-	v.ChangedParams(vagAddr == prevVagAddr);
+	v.loop = loop != 0;
+	v.playing = true;
```
Comment (unknownbrackets) [inline Core/HLE/sceSas.cpp:367]: I think we should still care about `on`?  You can key on or off a voice before or after setting the VAG, I'm pretty sure tests do show that already...

-[Unknown]

Comment (hrydgard) [reply Core/HLE/sceSas.cpp:367]: Oh yeah, I suppose so. I'll add that in.

**Pull request merged**